src/pkg/services/m365/api/groups.go

```diff
@@ -154,6 +155,75 @@ func (c Groups) GetByID(
 	return group, nil
 }
 
+// GetAllSites gets all the sites that belong to a group. This is
+// necessary as private and shared channels gets their on individual
+// sites. All the other channels make use of the root site.
+func (c Groups) GetAllSites(
```

Comment:
nit request: for future PRs the api layer additions can be easily separated into their own change, and the consumption of them split into follow-ups, if you would kindly separate those changes for easier reviews.